DirectProgramming/DPC++FPGA/Tutorials/Features/private_copies/src/private_copies.cpp

```diff
@@ -47,20 +49,18 @@ void PartialSumWithShift(const device_selector &selector, const IntArray &array,
       accessor accessor_array(buffer_array, h, read_only);
       accessor accessor_result(buffer_result, h, write_only, noinit);
 
-      h.single_task<Kernel<concurrency>>([=]() [[intel::kernel_args_restrict]] {
+      h.single_task<Kernel<num_copies>>([=]() [[intel::kernel_args_restrict]] {
         int r = 0;
 
-        // At most concurrency iterations of the outer loop will be
-        // active at one time.
-        // This limits memory usage, since each iteration of the outer
-        // loop requires its own copy of a1.
-        [[intel::max_concurrency(concurrency)]] for (size_t i = 0; i < kMaxIter;
-                                                     i++) {
-          int a1[kSize];
-          for (size_t j = 0; j < kSize; j++)
-            a1[j] = accessor_array[(i * 4 + j) % kSize] * shift;
-          for (size_t j = 0; j < kSize; j++) r += a1[j];
+        for (size_t i = 0; i < kMaxIter; i++) {
+          [[intel::private_copies(num_copies)]] int a[kSize];
```

Comment:
You may need a short comment before this line. Something like:
```c++
// Request num_copies private copies for array a. This limits the concurrency of the outer loop to num_copies and also limits the memory use of a.
[[intel::private_copies(num_copies)]] int a[kSize];
```
You may have to run clang-format on this comment though :)